### ISSUE TITLE
ADVML Week 1 Project review

### ISSUE THREAD

**Issue by bonareri** [Tactevil/second-hand-cars#1]
## ADVML Week 1 Project review

You demonstrated a good understanding of the data preparation and modelling workflow. Your feature engineering was thoughtful, you correctly excluded `Price per Mile` from the modelling features to avoid target leakage, and your preprocessing was organized using a Pipeline. You also compared multiple regression models and used GridSearchCV for model tuning.

However, **dimensionality reduction was not completed to the level required by the project**. The PCA section was mostly limited to defining a PCA pipeline, without demonstrating the actual reduction, retained components, explained variance, or comparing PCA and non-PCA model performance.

### 1. PCA and Dimensionality Reduction

This is the main area that needs improvement. You defined:

```python
PCA(n_components=0.95)
```

but did not report the actual PCA results.

The project requires you to apply PCA to reduce the feature space while retaining most of the variance. You should show:

* The number of features before PCA.
* The number of components retained.
* The percentage of variance retained.
* How much the feature space was reduced.
* Model performance with PCA versus without PCA.

Simply creating the PCA pipeline is not enough to demonstrate the dimensionality-reduction process.

### 2. Correlation Analysis

You calculated correlations, but you did not actually demonstrate the removal of highly correlated features as required by the project. You should define a correlation threshold and identify the features removed based on that threshold.

### 3. Outlier Handling

You winsorized `Mileage` and `Price` before the train-test split. The percentile thresholds should be determined using the training data only so that information from the test set does not influence preprocessing.

### 4. Model Evaluation

You compared several models, which is good. However, the PCA models were not included in the reported model results, so you cannot determine whether dimensionality reduction helped or hurt performance.

Your GridSearchCV section also needs to be followed by evaluation of the tuned model on the test set.

### 5. Price per Mile

You correctly excluded `Price per Mile` from the modelling features, which is important because it is calculated directly using the target `Price`.

However, be careful when interpreting its correlation with Price, since the feature itself contains Price.

### 6. Missing Values

You filled missing `Accessories` values with `"None"`. This assumes that a missing value means the car has no accessories. Explain and justify this assumption.

### 7. Lasso Warning

Your Lasso model produced a convergence warning. Address this before relying on its results, for example by increasing `max_iter` or adjusting the regularization settings.

### 8. Conclusion

Your conclusion that the models are learning "no useful signal" and that the predictions are "effectively noise" is too strong.

A better interpretation would be that the tested features and models **did not achieve performance better than the mean baseline on this dataset**. This is supported by the negative R² values.

## Score

**91/125 — 72.8%**

**Pass**

You have a solid foundation in preprocessing, feature engineering, and model evaluation, and you handled the `Price per Mile` leakage correctly. The main weakness is that the assignment's dimensionality-reduction requirements were only partially completed. In particular, PCA needs to be properly applied, analysed, and compared against the non-PCA approach.
